osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -212,6 +212,47 @@ public async Task<TModel> Import(string path, CancellationToken cancellationToke
             return import;
         }
 
+        /// <summary>
+        /// Import one <typeparamref name="TModel"/> from a <see cref="Stream"/>.
+        /// </summary>
+        /// <param name="stream">The stream to import files from.</param>
+        /// <param name="filename">The filename of the archive being imported.</param>
+        public async Task Import(Stream stream, string filename)
```

Comment:
Honestly I'd probably look into making

```csharp
public Task<TModel> Import(ArchiveReader archive, CancellationToken cancellationToken = default)
```

public, create the archive one level higher and just use that overload instead. There's a difference in that the initialisation notification won't fire right now, but maybe that logic should be moved there anyway. Silent imports aren't the best UX.

Reply:
I'm failing to understand the intent behind using this overload instead, and looking at the places which use this overload, not sure about moving import status notification logic there  (not sure, maybe move that logic one level higher aswell?). 

As a matter of fact I closed my game a few times while testing skin importing thinking that something went wrong whereas the skin was just taking a bit of time to import, and I hadn't added any import status notification yet,  so signaling the user the import is running is definitely needed, especially on mobile devices which may take a bit more of time to process files.

Reply:
Mostly I don't like the fact that the notification logic is partially duplicated inside the new overload, and that the stream+filename combo is passed down sort of unnecessarily. If you find a better way to address any of these then go ahead.

Reply:
ah, I see.  I that case , I think the common notification initialization code could be moved to a protected method , something like `CreateImportNotification()` and then the overload you mentionned earlier could accept optionally a `ProgressNotification` that it would update according to the import state.

Reply:
marking this as resolved, as #11115 handles this in a nice way.